File: documentation/manual/working/javaGuide/main/cache/code/javaguide/cache/inject/Application.java
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
 */
package javaguide.cache.inject;
//#inject
import play.cache.*;
import play.mvc.*;

import javax.inject.Inject;

public class Application extends Controller {

    private CacheApi cache;

    @Inject
    public Application(CacheApi cache) {
        this.cache = cache;
    }

    // ...
}
//#inject

File: documentation/manual/working/javaGuide/main/forms/code/javaguide/forms/controllers/Application.scala
/*
 * Copyright (C) Lightbend Inc. <https://www.lightbend.com>
 */

package javaguide.forms.controllers

import javax.inject.Inject

import play.api.mvc._

class Application @Inject() (components: ControllerComponents) extends AbstractController(components) {
  def submit = Action(Ok)
}
